pilot/pkg/xds/eds.go

```diff
@@ -183,36 +187,43 @@ func (s *DiscoveryServer) edsCacheUpdate(shard model.ShardKey, hostname string,
 		Namespace: namespace,
 	}: {}})
 	ep.mutex.Unlock()
-	if features.SendUnhealthyEndpoints && pushType == IncrementalPush {
+	if features.SendUnhealthyEndpoints {
 		// Check if new Endpoints are ready to be pushed. This check
 		// will ensure that if a new pod comes with a non ready endpoint,
 		// we do not unnecessarily push that config to Envoy.
 		// Please note that address is not a unique key. So this may not accurately
 		// identify based on health status and push too many times - which is ok since its an optimization.
 		emap := make(map[string]*model.IstioEndpoint, len(oldIstioEndpoints))
+		// Add new endpoints only if they are ever ready once to shards
+		// so that full push does not send them from shards.
+		newIstioEndpoints := []*model.IstioEndpoint{}
 		for _, oie := range oldIstioEndpoints {
 			emap[oie.Address] = oie
```

Comment:
Does not this matter, there can be many same address istioendpoints for multi port svc

Reply:
If it exists in old (even with single port) we will push if its health status changes.